platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy

```diff
@@ -414,6 +414,22 @@ class JavaCompileProblemsIntegrationTest extends AbstractIntegrationSpec impleme
         }
     }
 
+    def "invalid flags should be reported as problems"() {
+        given:
+        writeJavaCausingTwoCompilationWarnings("Foo")
+        buildFile << "tasks.compileJava.options.compilerArgs += ['-invalid-flag']"
+
+        when:
+        fails("compileJava")
+
+        then:
+        verifyAll(receivedProblem(0)) {
+            severity == Severity.ERROR
+            fqid == 'compilation:java:initialization-failed'
+            details == 'error: invalid flag: -invalid-flag'
```

Comment:
❓ Should we also add assertion for the exception here?

Reply:
Good point, it shouldn't hurt to check if the message is the same.

Reply:
Added